src/fractals/newtons_method.rs

```diff
@@ -0,0 +1,372 @@
+            &self.params.image_specification,
+            self.params.histogram_bin_count as u32,
+            self.histogram.clone(),
+        );
+        self.cdf.reset(&self.histogram);
+
+        for (color_table, inner_map) in self.color_maps.iter_mut().zip(self.inner_color_maps.iter())
+        {
+            reset_color_map_lookup_table_from_cdf(color_table, &self.cdf, inner_map);
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum SystemType {
+    RootsOfUnity(Box<RootsOfUnityParams>), // number of roots == root_colors_rgb.len()
+}
+
+impl<F> SpeedOptimizer for NewtonsMethodRenderable<F>
+where
+    F: ComplexFunctionWithSlope,
+{
+    type ReferenceCache = CommonParams;
+    fn reference_cache(&self) -> CommonParams {
+        self.params.clone()
+    }
+
+    fn set_speed_optimization_level(&mut self, level: f64, cache: &Self::ReferenceCache) {
+        self.params
+            .render_options
+            .set_speed_optimization_level(level, &cache.render_options);
+
+        // TODO:  finish implementing this method to improve responsive performance in explore.
+    }
+}
+
+impl<F> Renderable for NewtonsMethodRenderable<F>
+where
+    F: ComplexFunctionWithSlope + Sync + Send,
+{
+    type Params = CommonParams;
+    fn image_specification(&self) -> &ImageSpecification {
+        &self.params.image_specification
+    }
+
+    fn render_options(&self) -> &RenderOptions {
+        &self.params.render_options
+    }
+
+    fn set_image_specification(&mut self, image_specification: ImageSpecification) {
+        self.params.image_specification = image_specification;
+        self.update_color_map();
+    }
+
+    fn render_point(&self, point: &[f64; 2]) -> image::Rgb<u8> {
+        let result =
+            match self.newton_rhapson_iteration_sequence(Complex64::new(point[0], point[1])) {
+                Some(res) => res,
+                None => {
+                    return image::Rgb(self.params.cyclic_attractor_color_rgb);
+                }
+            };
+
+        // Use the solution to select the correct color map for this point:
+        let color_map_index = self.system.root_index(result.soln) % self.color_maps.len();
+        self.color_maps[color_map_index].compute_pixel(result.smooth_iteration_count)
+    }
+
+    fn write_diagnostics<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        self.histogram.display(writer)?;
+        self.cdf.display(writer)?;
+        std::io::Result::Ok(())
+    }
+
+    fn params(&self) -> &Self::Params {
+        &self.params
+    }
+}
+
+// Renders a Newton's method fractal based on the provided parameters.
+pub fn render_newtons_method(
+    params: &NewtonsMethodParams,
+    file_prefix: FilePrefix,
+) -> Result<(), Box<dyn std::error::Error>> {
+    // This indirection step is important for performance -- here we unwrap all of the `dyn` pointers
+    // and implement all of the inner render loops using generics for performance.
```

Comment:
Same concept here as with the Julia set and Mandelbrot set (both are rendered via the QuadraticMap generic class), but I moved it a level down so that we don't need to expose the different sub-types of the fractal at the top-level match in the main functions.